packages/kbn-unified-data-table/__mocks__/services.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import { of } from 'rxjs';
+import { uiActionsPluginMock } from '@kbn/ui-actions-plugin/public/mocks';
+import { expressionsPluginMock } from '@kbn/expressions-plugin/public/mocks';
+import { chromeServiceMock, coreMock } from '@kbn/core/public/mocks';
+import { chartPluginMock } from '@kbn/charts-plugin/public/mocks';
+import { fieldFormatsMock } from '@kbn/field-formats-plugin/common/mocks';
+import type { Storage } from '@kbn/kibana-utils-plugin/public';
+import { LocalStorageMock } from './local_storage_mock';
+import { IUiSettingsClient, ToastsStart } from '@kbn/core/public';
+import { DataViewFieldEditorStart } from '@kbn/data-view-field-editor-plugin/public';
+import { dataPluginMock } from '@kbn/data-plugin/public/mocks';
+
+export function createServicesMock() {
+  const expressionsPlugin = expressionsPluginMock.createStartContract();
+
+  expressionsPlugin.run = jest.fn(() =>
+    of({
+      partial: false,
+      result: {
+        rows: [],
+      },
+    })
+  ) as unknown as typeof expressionsPlugin.run;
+
+  const corePluginMock = coreMock.createStart();
+
+  const uiSettingsMock: Partial<typeof corePluginMock.uiSettings> = {
+    get: jest.fn(),
+    isDefault: jest.fn((key: string) => {
+      return true;
+    }),
+  };
+
+  corePluginMock.uiSettings = {
+    ...corePluginMock.uiSettings,
+    ...uiSettingsMock,
+  };
+
+  const theme = {
+    theme$: of({ darkMode: false }),
+  };
+
+  corePluginMock.theme = theme;
+
+  const dataPlugin = dataPluginMock.createStartContract();
+
+  return {
+    core: corePluginMock,
+    charts: chartPluginMock.createSetupContract(),
+    chrome: chromeServiceMock.createStartContract(),
+    history: () => ({
+      location: {
+        search: '',
+      },
+      listen: jest.fn(),
+    }),
+    fieldFormats: fieldFormatsMock,
+    filterManager: jest.fn(),
+    inspector: {
+      open: jest.fn(),
+    },
+    uiActions: uiActionsPluginMock.createStartContract(),
+    uiSettings: uiSettingsMock as IUiSettingsClient,
+    http: {
+      basePath: '/',
+    },
+    dataViewEditor: {
+      openEditor: jest.fn(),
+      userPermissions: {
+        editDataView: jest.fn(() => true),
+      },
+    },
+    dataViewFieldEditor: {
+      openEditor: jest.fn(),
+      userPermissions: {
+        editIndexPattern: jest.fn(() => true),
+      },
+    } as unknown as DataViewFieldEditorStart,
+    theme,
+    storage: new LocalStorageMock({}) as unknown as Storage,
```

Comment:
Nit: I dont' think we need `LocalStorageMock` (which is basically an implementation of `Storage`):
```suggestion
    storage: {
      clear: jest.fn(),
      get: jest.fn(),
      set: jest.fn(),
      remove: jest.fn(),
    },
```
Then we can remove the `LocalStorageMock` altogether.